### DISCUSSION TITLE
Sandbox mode - Allow a user to build or import a GameState and step live

### DISCUSSION THREAD

**Discussion by BoldBigflank** [BattlesnakeOfficial/feedback#127] [Feature Requests]
When I get my snake to a point in the game where they make a poor choice, I want to be able to start at that point and repeat it until my updated code makes the right choice.

One way of doing this would be to make a sort of "sandbox" that I could draw snakes/food/hazards onto, then hit a button to ask my snake what it would do there. Then, any game I see on the website could have an "open in sandbox mode" button to move that game state over to the sandbox page.

It might only be limited to games with all public snakes, since it would be asking those to play too.

**Comment by coreyja** [comment]
Not an official solution, but community member Nettogrof has put together a board generator that lets you do almost exactly this. I guess the bit its missing is that it can only run against YOUR snake, not all the opponents as well

You can 'draw' a board and then run it against your snake! You can also import and export the game state to JSON!

https://nettogrof.github.io/battle-snake-board-generator/

**Reply by BoldBigflank** [reply]
Thanks for that. It seems like a start, but I think that the url is hard coded and not responsive to what you type into the url field. I might try my hand at making my own, perhaps as a browser extension for easy importing.

**Reply by coreyja** [reply]
Oh ya I know Nettogrof has implemented a proxy to get around CORS issues, and it looks like the proxy might be down right now

Paging @Nettogrof for the FYI

**Reply by andrewdmaclean** [reply]
👍
